### PR TITLE
feat(analytics): allow reporting without environment

### PR DESCRIPTION
**Pull request by tomasklim**
## Description
 
Environment in analytics is now optional as we do not need it for analytics in connect

## Related Issue

https://github.com/trezor/trezor-suite/issues/5397

### CODE REVIEW

**Comment by matejkriz** [comment]
The environment could become useful for connect popup too if we add support for popup in modal instead of just new tab, right @mroz22? But for sure it's ok to make it optional for now.

**Review by mroz22 — APPROVED** [review]

**Comment by mroz22** [comment]
maybe, its ok now

File: packages/analytics/src/tests/analytics.test.ts [hunk 1]
```diff
@@ -57,6 +57,40 @@ describe('analytics', () => {
...
+        });
+        global.fetch = jest.fn().mockImplementation(() => mockFetchPromise);
+        // @ts-expect-error
+        jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise);
+
+        const timestamp = new Date().getTime();
+        jest.spyOn(Date, 'now').mockImplementation(() => timestamp);
+        jest.spyOn(console, 'error').mockImplementation(() => {});
+
+        const app = 'connect';
```
Comment (matejkriz) [inline packages/analytics/src/tests/analytics.test.ts:72]: according to the `App` type `'suite' | 'connect-popup'`
```suggestion
        const app = 'connect-popup';
```

**Comment by tomasklim** [comment]
> The environment could become useful for connect popup too if we add support for popup in modal instead of just new tab, right @mroz22? But for sure it's ok to make it optional for now.

https://github.com/trezor/webwallet-data/pull/43#issuecomment-1268229951

File: packages/analytics/src/tests/analytics.test.ts (hunk 1 above)
Comment (tomasklim) [reply packages/analytics/src/tests/analytics.test.ts:72]: fixed, squashed, rebased

**Pull request merged**